handler/implementations/passThrough.go

```diff
@@ -483,3 +523,34 @@ func (h *PassThrough) GetResourceMutex(n domain.IOnodeIface) *sync.Mutex {
 func (h *PassThrough) SetService(hs domain.HandlerServiceIface) {
 	h.Service = hs
 }
+
+// checkProcAndSysRemount checks if the nsenter agent deployed by the passthrough handler
+// should remount procfs and sysfs.
+func checkProcAndSysRemount(n domain.IOnodeIface) (bool, bool, uint32) {
+	var (
+		mountSysfs  bool
+		mountProcfs bool
+		cloneFlags  uint32
+	)
+
+	// The nsenter agent will enter the namespaces of the container (except the
+	// mount ns to avoid a recursion with sysbox-fs). However, when accessing
+	// files under /proc or /sys, the agent needs to remount these as otherwise
+	// they won't pick up the container's assigned resources (e.g., net devices,
+	// etc). We do this by having the nsenter agent in a new mount namespace so
+	// as to not mess up mounts in the host or in the container.
+
+	if strings.HasPrefix(n.Path(), "/sys/") {
+		mountSysfs = true
+	}
+
+	if strings.HasPrefix(n.Path(), "/proc/") {
```

Comment:
Why also for `procfs`? Have you identified a use-case that requires it?

Reply:
I've not identified a use case for procfs yet, but I figured it would be good to add in case a handler wants the nsenter agent to mount it. Note that the code I added won't mount procfs; it's up to the handlers to set the mountProcfs flag in the nsenter request and no handler currently does it.

Reply:
This is the main concern I have with this change: performance penalty, as we are now explicitly requesting a procfs and a sysfs mount for every incoming file operation invoking the `passthrough` handler. If we end up noticing a significant hit (which I don't really anticipate), we could always do some tuning by narrowing down the file-system paths on which we do the mount operation (e.g., there's probably no need to mount procfs if the `passthrough` handler is looking at a `/proc/sys/kernel` resource).